exec/bigmachine.go

```diff
@@ -168,7 +181,13 @@ func (b *bigmachineExecutor) manager(i int) *machineManager {
 			maxLoad = 0
 		}
 		b.managers[i] = newMachineManager(b.b, b.params, b.status, b.sess.Parallelism(), maxLoad, b.worker)
-		go b.managers[i].Do(backgroundcontext.Get())
+
+		b.managersWG.Add(1)
+		go func() {
+			defer b.managersWG.Done()
```

Comment:
This seems a gratuitous use of defer. It doesn't make the code cleaner or more concise: just put it at the end of the func(){}.